sourcegraph/sourcegraph
# Search: Add the full UI mode for the search result page

Closes https://github.com/sourcegraph/sourcegraph/issues/40360
Based on https://github.com/sourcegraph/sourcegraph/pull/40138

## Background 

This PR adds search aggregation full UI mode for the search result page. This PR doesn't imply connecting UI and the real datasets from the backend. It will be addressed later in this issue https://github.com/sourcegraph/sourcegraph/issues/40306

Aggregation UI modes work through URL query param, after you set aggregation mode (or aggregation type) it should be persistent in URL this allows you to share the same aggregation mode and aggreagtion UI that you can see on you screen as you copy URL to share it. 

| Simple UI mode  | Standard UI |
| ------------- | ------------- |
| <img width="1159" alt="Screenshot 2022-08-17 at 11 23 58" src="https://user-images.githubusercontent.com/18492575/185110774-179669cc-8edc-4643-ba5c-75c0704396aa.png"> | <img width="1159" alt="Screenshot 2022-08-17 at 11 58 51" src="https://user-images.githubusercontent.com/18492575/185110831-e4f8e7e0-1575-4762-9e1b-5f2f04d3ea1a.png"> |

## Todo 
- [x] Restructure aggregation UI models and components for future reuse in the full UI mode
- [x] Implement the full UI mode screen 
- [x] Connect the sidebar, and the full UI mode states via URL query params
 
## Test plan
- Make sure that the full mode UI looks visually correct on desktop, tablet, and mobile layouts
- Make sure that the full UI mode persists state in URL, and after refresh, you still see the full mode UI by default (if you don't close it before refresh)



## App preview:

- [Web](https://sg-web-vk-add-aggregation-full-mode-ui.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-drbafirzqo.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
